Example/WalletApp/ApplicationLayer/SceneDelegate.swift

```diff
@@ -0,0 +1,44 @@
+import UIKit
+import Auth
+import WalletConnectPairing
+
+final class SceneDelegate: UIResponder, UIWindowSceneDelegate {
+    var window: UIWindow?
+
+    private let app = Application()
+
+    private var configurators: [Configurator] {
+        return [
+            MigrationConfigurator(app: app),
+            ThirdPartyConfigurator(),
+            ApplicationConfigurator(app: app),
+            AppearanceConfigurator()
+        ]
+    }
+    
+    func application(_ application: UIApplication, configurationForConnecting connectingSceneSession: UISceneSession, options: UIScene.ConnectionOptions) -> UISceneConfiguration {
+          let sceneConfig: UISceneConfiguration = UISceneConfiguration(name: nil, sessionRole: connectingSceneSession.role)
+          sceneConfig.delegateClass = SceneDelegate.self
+          return sceneConfig
+      }
+
+    func scene(_ scene: UIScene, willConnectTo session: UISceneSession, options connectionOptions: UIScene.ConnectionOptions) {
+        guard let windowScene = (scene as? UIWindowScene) else { return }
+
+        window = UIWindow(windowScene: windowScene)
+        window?.makeKeyAndVisible()
+        
+        app.uri = connectionOptions.urlContexts.first?.url.absoluteString.replacingOccurrences(of: "walletapp://wc?uri=", with: "")
```

Comment:
What the idea of doing that?

Reply:
The logic behind is to get the **wc uri** if app was opened via deeplink and to be able to handle within any screen in the wallet app.
Like in the current UI we have a welcome screen which is not responsible for setting any kind of subscriptions and performing some requests.
So when I save **uri** to the **app**, I can trigger handling in the next screen - **WalletView** or any other.